libs/postgres_ffi/src/xlog_utils.rs

```diff
@@ -154,18 +156,23 @@ mod timestamp_conversions {
         }
     }
 
+    /// TODO: this function should be deprecated, likely malicious input could cause the .unwrap() to panic.
     pub fn from_pg_timestamp(time: TimestampTz) -> SystemTime {
+        try_from_pg_timestamp(time).unwrap()
+    }
```

Comment:
@arpad-m what was the basis for using the .exect() when you implemented get_lsn_by_timestamp endpoint?

Reply:
You mean f842b22b907d8682b6f900a8d7724a6aa4b52e77 ? It's almost a year ago but I suppose back then I thought that if the `SystemTime -> TimestampTz` conversion uses a panic, the `TimestampTz -> SystemTime` conversion can do so too. The provided `SystemTime` can also be user chosen I think.

Aren't panics safe though? They will print a big backtrace about something in the http handler but life will go on, no? Unless you set `panic=abort`, which some people do, but I'm not a fan of that myself.

I'm not objecting to the changes in this PR, but I would say that trying to make the pageserver never panic from malicious input is extremely tough and it's extremely hard to get it right. Rust is a good language to avoid/reduce UB, with all the security implications this has, but Rust still makes it super easy to introduce potentially panicking locations.

Reply:
>  I thought that if the SystemTime -> TimestampTz conversion uses a panic, the TimestampTz -> SystemTime conversion can do so too

😂

I figured.

> I'm not objecting to the changes in this PR, but I would say that trying to make the pageserver never panic from malicious input is extremely tough and it's extremely hard to get it right. 

If it's user-controlled input then we should not be panicking.

(Admittedly that rule is broken in many places and not written down, will take a note to have a small session about this during the offsite)

Reply:
https://www.notion.so/neondatabase/Storage-Lisbon-a340ba91b30f4d0f9c4ce676559df37e?pvs=4#354ca8b49bfb41c5bb070ddbb4dbc98a